src/audio/rtnr/rtnr.c

```diff
@@ -0,0 +1,554 @@
+		break;
+	default:
+		comp_err(dev, "rtnr_cmd_set_data() error: invalid command %d", cdata->cmd);
+		ret = -EINVAL;
+		break;
+	}
+	return ret;
+}
+
+static int rtnr_cmd_set_data(struct comp_dev *dev,
+			    struct sof_ipc_ctrl_data *cdata)
+{
+	int ret = 0;
+
+	switch (cdata->cmd) {
+	case SOF_CTRL_CMD_BINARY:
+		comp_info(dev, "rtnr_cmd_set_data(), SOF_CTRL_CMD_BINARY");
+		break;
+	default:
+		comp_err(dev, "rtnr_cmd_set_data() error: invalid command %d", cdata->cmd);
+		ret = -EINVAL;
+		break;
+	}
+	return ret;
+}
+
+/* used to pass standard and bespoke commands (with data) to component */
+static int rtnr_cmd(struct comp_dev *dev, int cmd, void *data,
+		   int max_data_size)
+{
+	struct sof_ipc_ctrl_data *cdata = data;
+	int ret = 0;
+
+	comp_info(dev, "rtnr_cmd()");
+
+	switch (cmd) {
+	case COMP_CMD_SET_DATA:
+		ret = rtnr_cmd_set_data(dev, cdata);
+		break;
+	case COMP_CMD_GET_DATA:
+		ret = rtnr_cmd_get_data(dev, cdata, max_data_size);
+		break;
+	}
+
+	return ret;
+}
+
+static int rtnr_trigger(struct comp_dev *dev, int cmd)
+{
+	comp_info(dev, "rtnr_trigger() cmd: %d", cmd);
+
+	return comp_set_state(dev, cmd);
+}
+
+/* copy and process stream data from source to sink buffers */
+static int rtnr_copy(struct comp_dev *dev)
+{
+	struct list_item *blist;
+	struct comp_buffer *sink;
+	struct comp_buffer *source;
+	int frames;
+	int num_blocks;
+	int sink_bytes;
+	int source_bytes;
+	const struct audio_stream *sources_stream[RTNR_MAX_SOURCES] = { NULL };
+	struct comp_data *cd = comp_get_drvdata(dev);
+	struct comp_buffer *sources[RTNR_MAX_SOURCES] = { NULL };
+	uint32_t flags = 0;
+	int num_sources = 0;
+
+	comp_dbg(dev, "rtnr_copy()");
+
+	/* align source streams with their respective configurations */
+	list_for_item(blist, &dev->bsource_list) {
+		source = container_of(blist, struct comp_buffer, sink_list);
+		buffer_lock(source, &flags);
+		if (source->source->state == dev->state) {
+			if (num_sources < RTNR_MAX_SOURCES) {
+				sources[num_sources] = source;
+				sources_stream[num_sources] = &source->stream;
+				num_sources++;
+			} else {
+				buffer_unlock(source, flags);
+			}
+		} else {
+			buffer_unlock(source, flags);
+		}
+	}
+
+	/* check if there are any sources active */
+	if (num_sources == 0 || num_sources > 2)
+		return -EINVAL;
+
+	#if RTNR_ENABLE_PROCESSING
+	/* put dummy data into output queue*/
+	int nch = sources_stream[0]->channels;
+	int copy_size = BLOCKLENGTH*nch*sizeof(int32_t);
+
+	RTKMA_API_First_Copy((void *)&cd->rtk_agl, copy_size);
+	#endif // RTNR_ENABLE_PROCESSING
+
+	sink = list_first_item(&dev->bsink_list, struct comp_buffer, source_list);
+	buffer_lock(sink, &flags);
+	frames = audio_stream_avail_frames(sources_stream[0], &sink->stream);
+	buffer_unlock(sources[0], flags);
+
+	/* Process integer multiple of RTNR internal block length */
+	num_blocks = frames / RTNR_BLK_LENGTH;
+	frames = num_blocks * RTNR_BLK_LENGTH;
+
+	if (cd->count) {
```

Comment:
remove this counting logic please, this should be a simple debug level print. Copy logs should never print at info level.

Reply:
The counting logic has be removed.